R/PackageFunctionReporter.R

```diff
@@ -185,40 +183,98 @@ FunctionReporter <- R6::R6Class(
 
             log_info(sprintf('Constructing network representation...'))
 
-            # foodweb will output a warning for "In par(oldpar) : calling par(new=TRUE) with no plot" all the time.
-            # does not seem to be an issue
-            funcMap <- suppressWarnings({
-                mvbutils::foodweb(
-                    where = paste("package"
-                                  , self$pkg_name
-                                  , sep = ":")
-                    ,
-                    plotting = FALSE
-                )
-            })
+            # Get table of edges between functions
+            edgeDT <- .get_edges(pkg_name = self$pkg_name)
 
             log_info("Done constructing network representation")
 
-            # Function Connections: Edges
-            edges <- data.table::melt(
-                data.table::as.data.table(funcMap$funmat, keep.rownames = TRUE)
-                , id.vars = "rn"
-            )[value != 0]
-
-            # Formatting
-            edges[, value := NULL]
-            edges[, SOURCE := as.character(variable)]
-            edges[, TARGET := as.character(rn)]
-            edges[, variable := NULL]
-            edges[, rn := NULL]
-            data.table::setcolorder(edges, c('SOURCE', 'TARGET'))
-
-            # If no edges, return empty data.table
-            if (nrow(edges) == 0) {
-                return(data.table::data.table(SOURCE = character(), TARGET = character()))
-            }
-
-            return(edges)
+            return(edgeDT)
         }
     )
 )
+
+#' @importFrom data.table rbindlist
+.get_edges <- function(pkg_name){
+
+    # find all functions in this package
+    obj_names <- ls(sprintf("package:%s", pkg_name))
+
+    # create a custom environment w/ this package's contents
+    pkg_env <- loadNamespace(pkg_name)
+
+    # Filter to just function objects. This will now be a character
+    # vector full of function names
+    funs <- Filter(
+        f = function(x, p = pkg_env){is.function(get(x, p))}
+        , x = obj_names
```

Comment:
If you pass in `x = names(pkg_env)` instead of `obj_names` here, you'll get all of the internal non-exported functions for the package as well.

Reply:
Up to you if you want to address this in this PR or wait for a separate one. I think we'd maybe want to think about capturing the metadata about whether it's exported so that we can display that visually somehow too.

Reply:
oh nice, really elegant fix!

I want @bburns632 to weight in (here or on #79 ) with his feelings on the inclusion of internal stuff.

Good call that we'd probably want to keep it as additional context that could be shown visually

Reply:
I don't see a downside to including internal functions.  Is there a reason to distinguish between the two?

Reply:
nice! I think we should definitely include them, and then make an issue with "enhancement" label that talks about visually distinguishing them. We can discuss relative merits of adding that metadata on that issue.

I'll just include them as part of this PR (per Jay's recommendation)